chainreg/chainregistry.go

```diff
@@ -840,3 +873,31 @@ var (
 		},
 	}
 )
+
+// checkOutboundPeers checks the number of outbound peers connected to the
+// provided RPC client. If the number of outbound peers is below 6, a warning
+// is logged. This function is intended to ensure that the chain backend
+// maintains a healthy connection to the network.
+func checkOutboundPeers(client *rpcclient.Client) error {
+	peers, err := client.GetPeerInfo()
+	if err != nil {
+		return err
+	}
+
+	var outboundPeers int
+	for _, peer := range peers {
+		if !peer.Inbound {
+			outboundPeers++
+		}
+	}
+
+	if outboundPeers < DefaultMinOutboundPeers {
+		log.Warnf("The chain backend has an insufficient number "+
+			"of connected outbound peers (%d connected, expected "+
+			"minimum is %d) which can be a security issue. "+
+			"Connect to more trusted nodes manually if necessary.",
+			outboundPeers, DefaultMinOutboundPeers)
+	}
+
+	return nil
+}
```

Comment:
> consider enhancing it by returning a specific error when the number of outbound peers is critically low, allowing for more decisive action than just logging.

I don't think this is what we want since bitcoind should recover from this automatically.